pkg/bootstrap/platform/gcp_test.go

```diff
@@ -258,6 +260,34 @@ func TestGCPMetadata(t *testing.T) {
 	}
 }
 
+func TestGCPQuotaProject(t *testing.T) {
+	cases := []struct {
+		name         string
+		quotaProject string
+		wantFound    bool
+		wantProject  string
+	}{
+		{"no value set", "", false, ""},
+		{"value set", "234234323", true, "234234323"},
+	}
+	for _, v := range cases {
+		t.Run(v.name, func(tt *testing.T) {
+			shouldFillMetadata = func() bool { return true }
+			os.Setenv("GCP_QUOTA_PROJECT", v.quotaProject)
+			meta := NewGCP().Metadata()
+			val, found := meta[GCPQuotaProject]
+			if got, want := found, v.wantFound; got != want {
+				tt.Errorf("Metadata() returned an unexpected value for GCPQuotaProject; found value: %t, want %t", got, want)
+			}
+			if got, want := val, v.wantProject; got != want {
+				tt.Errorf("Incorrect value for GCPQuotaProject; got = %q, want = %q", got, want)
+			}
+			os.Unsetenv("GCP_QUOTA_PROJECT")
+			shouldFillMetadata = metadata.OnGCE
```

Comment:
Maybe wrap in defer right after the corresponding set calls for extra maintainability

Reply:
sg. done.